### PR TITLE
Adding ECHO support 

### PR DESCRIPTION
**Pull request by jsschreck**

### CODE REVIEW

**Comment by katiedagon** [comment]
This all looks good to me. @mariajmolina I wonder if we can connect this repo to [ReviewNB](https://www.reviewnb.com/) to more easily review notebooks.

File: ML-for-S2S/gan.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+seed: 1000
+save_loc: "/glade/work/schreck/repos/ML-for-S2S/ML-for-S2S"
```
Comment (kjmayer) [inline ML-for-S2S/gan.yml:2]: Probably want to update these paths (gust and curiosity)

File: s2sml/load_loss.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import torch, logging
+
+
+def load_loss(loss_type):
```
Comment (kjmayer) [inline s2sml/load_loss.py:4]: Is perceptual similarity going here or somewhere else? I might've missed this is the slack messages

File: environment.yml [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+  - yaml=0.2.5=h7f98852_2
+  - zeromq=4.3.4=h2531618_0
+  - zict=2.2.0=pyhd8ed1ab_0
+  - zlib=1.2.13=h166bdaf_4
+  - zstd=1.5.2=h3eb15da_6
+  - pip:
+    - segmentation-models-pytorch
+    - lpips
+    - piqa
+    - echo-opt
```
Comment (kjmayer) [inline environment.yml:349]: leaving a comment as a reminder to fix the pip install of echo :)

File: s2sml/torch_s2s_dataset.py [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+        
+    Returns:
+        list of cesm and era5 files for train/val/test sets.
+    
+    """
+    
+    def __init__(self, week, variable, norm='zscore', region='fixed',
+                 minv=None, maxv=None, mnv=None, stdv=None, lon0=None, lat0=None, dxdy=32,
+                 feat_topo=True, feat_lats=True, feat_lons=True, 
+                 startdt='1999-02-01', enddt='2021-12-31', homedir='/glade/scratch/molina/'):
```
Comment (kjmayer) [inline s2sml/torch_s2s_dataset.py:41]: @mariajmolina should we make the startdt and enddt based on seasons? 
Ex: `startdt='1999-11-01, enddt='2021-02-28'`

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from setuptools import setup
+
+setup(name="s2sml",
+      version="0.1",
+      description="Machine learning for s2s prediction",
+      author="Maria Molina, Katie Dagon, Kirsten Mayer, John Schreck",
```
Comment (kjmayer) [inline setup.py]:
```suggestion
      author="Maria Molina, Katie Dagon, Kirsten Mayer, John Schreck, Jhayron Perez",
```

File: ML-for-S2S/gan.yml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
```
Comment (kjmayer) [inline ML-for-S2S/gan.yml:3]: Some comments about what some of these variables mean would be really helpful :)

**Review by kjmayer — COMMENTED** [review]
Looks good - just more comments throughout about what some things mean/do would be helpful! Thanks for doing all of this!!

File: s2sml/load_loss.py (hunk 2 above)
Comment (jsschreck) [reply s2sml/load_loss.py:4]: In the initial commit it was only being used as a metric, but I will add it here! (I've yet to actually backprop with it, I will run some tests before updating)..

File: s2sml/torch_s2s_dataset.py (hunk 4 above)
Comment (mariajmolina) [reply s2sml/torch_s2s_dataset.py:41]: These are just default values -- feel free to test using seasons / dates you want to use. I am going to keep the full range because it is just bias correction in this application.

**Comment by mariajmolina** [comment]
> This all looks good to me. @mariajmolina I wonder if we can connect this repo to [ReviewNB](https://www.reviewnb.com/) to more easily review notebooks.

Added ReviewNB access

File: ML-for-S2S/gan.yml (hunk 1 above)
Comment (mariajmolina) [reply ML-for-S2S/gan.yml:2]: yes and we may want these to be our own paths for our own trials

File: environment.yml (hunk 3 above)
Comment (mariajmolina) [reply environment.yml:349]: @jsschreck was the pip install fixed for echo? if not, should we leave this out of the yml file for now?

File: s2sml/load_loss.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        return torch.nn.L1Loss()
+    if loss_type == "smooth":
+        return torch.nn.SmoothL1Loss()
+    if loss_type == "logcosh":
+        return LogCoshLoss()
+    if loss_type == "xtanh":
+        return XTanhLoss()
+    if loss_type == "xsigmoid":
+        return XSigmoidLoss()
+
```
Comment (mariajmolina) [inline s2sml/load_loss.py]:
```suggestion
    if loss_type == "huber":
        # When delta is set to 1, this loss is equivalent to SmoothL1Loss. In general, this loss differs from SmoothL1Loss by a factor of delta (AKA beta in Smooth L1).
        return torch.nn.HuberLoss(reduction='mean', delta=0.5)
```

Comment (mariajmolina) [reply s2sml/load_loss.py]: Would love to add huber here too if we can figure out an easy way to set delta

**Review by mariajmolina — COMMENTED** [review]
Looks good! Just one main question pending (re: echo-opt pip install for yml file)
